### PR TITLE
azhou - Course Details Button

### LINKED ISSUE #35: Button to route to placeholder page for SectionTable (F2)
# F2 pt1: Button to route to placeholder page (for SectionTable)

- [ ] There is a button, hopefully small, perhaps with the logo for "more information" (ⓘ or one of the alternatives explained [here](https://stackoverflow.com/a/69679332/13960329)) beside every course in the courses table (i.e. search results) both lectures and sections that takes you to the course details page with the correct quarter and enrollcode.

Part of F2 for issue https://github.com/ucsb-cs156-f23/proj-courses-f23-7pm-4/issues/18

### PR DESCRIPTION
**Pull request by apzhou0**
In this PR, we add a ⓘ button beside each course entry of the SectionsTable (used for the main search, search by Course Location History, and search by Instructor) that, when clicked, will bring the user to the Course Details page for that course. We also change the quarter from 'yyyyq' to 'qyy' format. 

Dokku Link: https://proj-courses-apzhou0-dev.dokku-12.cs.ucsb.edu/

To test, click the ⓘ button in the rightmost column of any course. It should work even if you are not logged in.

image

SectionsTable Storybook Link: https://ucsb-cs156-f23.github.io/proj-courses-f23-7pm-4/prs/46/storybook/?path=/docs/components-sections-sectionstable--empty
CourseDetailsPage Storybook Link: https://ucsb-cs156-f23.github.io/proj-courses-f23-7pm-4/prs/46/storybook/?path=/docs/pages-coursedetails-coursedetailspage--default

Closes #35

### CODE REVIEW

**Review by audoreven — APPROVED** [review]
LGTM!

**Review by EricM5 — APPROVED** [review]
LGTM

**Review by pconrad — CHANGES_REQUESTED** [review]
There is a bug that needs to  be fixed: if you do a course search without being logged in, the course details button doesn't work.  But it should.

I think the only thing you need to do in order to fix this is to move this code in `App.js`
```jsx
  <Route
              exact
              path="/coursedetails/:yyyyq/:enrollCd"
              element={<CourseDetailsPage />}
            />
```

Move it out of the block that's guarded by:

```
  {hasRole(currentUser, "ROLE_USER") && (
``` 

And into the group at the bottom, the routes that are available even when you are not logged in.

I think that should fix it.   Give it a try.

File: frontend/src/main/components/Sections/SectionsTable.js [hunk 1]
```diff
@@ -17,6 +18,13 @@ function getFirstVal(values) {
 export default function SectionsTable({ sections, canExpand = true }) {
   // Stryker enable all
```
Comment (pconrad) [inline frontend/src/main/components/Sections/SectionsTable.js:19]: This line should never have been allowed to be approved in this file.  Please remove it; it is unnecessary.  Stryker is enabled by default.

File: frontend/src/main/components/Sections/SectionsTable.js [hunk 2]
```diff
@@ -17,6 +18,13 @@ function getFirstVal(values) {
...
   // Stryker disable BooleanLiteral
```
Comment (pconrad) [inline frontend/src/main/components/Sections/SectionsTable.js:20]: This line should never have been allowed to be in this file; it is likely to cover up a number of real testing issues.  Please remove it.  If that results in mutants that survive, we can look at whether to use a more surgical approach to a stryker exception, or just add the tests that are needed.

Comment (pconrad) [inline frontend/src/main/components/Sections/SectionsTable.js:20]:
```suggestion

```

File: frontend/src/main/components/Sections/SectionsTable.js (hunk 1 above)
Comment (pconrad) [inline frontend/src/main/components/Sections/SectionsTable.js:19]:
```suggestion

```

File: frontend/src/main/pages/CourseDetails/CourseDetailsPage.js [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+import CourseDetailsTable from "main/components/Courses/CourseDetailsTable";
 import { useParams } from "react-router-dom";
+import { qyyToYyyyq } from "main/utils/quarterUtilities.js";
 
 export default function CourseDetailsPage() {
   const { qyy, enrollCd } = useParams();
+  const yyyyq = qyyToYyyyq(qyy.toString());
+
+  const { data: personalSection } = useBackend(
+    // Stryker disable all : hard to test for query caching
```
Comment (pconrad) [inline frontend/src/main/pages/CourseDetails/CourseDetailsPage.js:12]: This needs a matching `// Stryker restore all`

File: frontend/src/main/pages/CourseDetails/CourseDetailsPage.js [hunk 4]
```diff
@@ -1,15 +1,28 @@
...
+    [`/api/sections/sectionsearch?qtr=${yyyyq}&enrollCode=${enrollCd}`],
+    {
+      // Stryker disable next-line all : GET is the default, so changing this to "" doesn't introduce a bug
+      method: "GET",
+      url: `/api/sections/sectionsearch?qtr=${yyyyq}&enrollCode=${enrollCd}`,
+    },
+  );
```
Comment (pconrad) [inline frontend/src/main/pages/CourseDetails/CourseDetailsPage.js:20]:
```suggestion
// Stryker restore all
```

**Review by pconrad — CHANGES_REQUESTED** [review]
Several issues to take a look at.

**Review by Bennyyyyy — APPROVED** [review]
LGTM

**Pull request merged**